### DISCUSSION TITLE
Any questions about project

### DISCUSSION THREAD

**Discussion by Klorestz** [pedroSG94/RootEncoder#1335] [Q&A]
Hello, Pedro. I'm a newbie, and I have a few questions: which specific IP and port should I use? I've tried various IP addresses, including the WiFi IP address listed in the emulator's WiFi settings and on a real device, but I keep getting a 'Retry' message and can't start the server.

I also have a small question regarding VPN: if the VPN server is enabled, which IP address should I use, the local one or the VPN server's? Can multiple devices connect to a running stream simultaneously?

**Comment by pedroSG94** [comment]
Hello,

First of all you need to understand that this project **send** video/audio to other device where you have installed a media server waiting to receive that data. It is not a server
So you will need to install a media server in a computer. You have servers here:
https://github.com/pedroSG94/RootEncoder/wiki/Media-servers-tested
Or you can find your own server.

After that, the response is easy. The IP is your computer IP where you installed the media server and the port is the port configured by your media server.
About the VPN, it is a question for your system admin that configure the VPN. Any special environments like VPN or firewall should be configured correctly by yourself.

**Reply by Klorestz** [reply]
Thank you for your answer, what about miltiple devices connected? Are there any problems with this?

**Reply by pedroSG94** [reply]
What do you mean? With this library you are pushing data to a media server. This is a per to per connection, you haven't multiple devices to connect.
The media server is who need support receive multiple pushers

**Reply by Klorestz** [reply]
I see you have an RTSP-server repository, can it be used as a server? https://github.com/pedroSG94/RTSP-Server

**Comment by pedroSG94** [comment]
Not at all.
This only send data to one or multiple RTSP players but doesn't accept receive data from a pusher

**Reply by Klorestz** [reply]
I have an application for Kotlin, with which I read RTSP streams through the VLC player. I want to add a feature so that the phone itself can become an RTSP server and can send an RTSP stream to the network, so that later through my application I can connect to this stream. Do I understand correctly that for such a task I can just use your implementation of the RTSP server?

**Reply by pedroSG94** [reply]
Yes, it is possible using that project. You can compile the example and test it

**Reply by Klorestz** [reply]
I was able to get the server up with your project and connect to it, but  the connection works for about 1-6 seconds, both video and audio are visible. But then the connection drops. With what it can be connected? I see this logs: 

Client disconnected
                                                                                                    java.net.SocketException: Socket closed
                                                                                                      at java.net.SocketInputStream.socketRead0(Native Method)
                                                                                                      at java.net.SocketInputStream.socketRead(SocketInputStream.java:118)
                                                                                                      at java.net.SocketInputStream.read(SocketInputStream.java:173)
                                                                                                      at java.net.SocketInputStream.read(SocketInputStream.java:143)
                                                                                                      at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:291)
                                                                                                      at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:355)
                                                                                                      at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:181)
                                                                                                      at java.io.InputStreamReader.read(InputStreamReader.java:180)
                                                                                                      at java.io.BufferedReader.fill(BufferedReader.java:172)
                                                                                                      at java.io.BufferedReader.readLine(BufferedReader.java:337)
                                                                                                      at java.io.BufferedReader.readLine(BufferedReader.java:403)
                                                                                                      at com.pedro.rtsp.rtsp.commands.CommandsManager.getResponse(CommandsManager.kt:237)
                                                                                                      at com.pedro.rtspserver.ServerCommandManager.getRequest(ServerCommandManager.kt:132)
                                                                                                      at com.pedro.rtspserver.ServerClient.run(ServerClient.kt:51)

**Reply by pedroSG94** [reply]
According with the logs the player side close the connection. Try with VLC in a PC to discard errors in the player side

**Reply by Klorestz** [reply]
I received the following VLC logs, sorry for such a number of questions, but you helped me a lot!

-- logger module started --
main: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
videotoolbox: Using Video Toolbox to decode 'h264'
videotoolbox: Using Video Toolbox to decode 'h264'
videotoolbox: vt cvpx chroma: 420v
-- logger module stopped --
﻿-- logger module started --
main: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
videotoolbox: Using Video Toolbox to decode 'h264'
main error: buffer deadlock prevented
videotoolbox: Using Video Toolbox to decode 'h264'
videotoolbox: vt cvpx chroma: 420v
live555 error: Failed to connect with rtsp://192.168.31.235:1935/
main error: connection failed: Connection refused
satip error: Failed to connect to RTSP server 192.168.31.235:1935
main error: connection failed: Connection refused
access_realrtsp error: cannot connect to 192.168.31.235:1935
live555 error: Failed to connect with rtsp://192.168.31.235:1935/
main error: connection failed: Connection refused
satip error: Failed to connect to RTSP server 192.168.31.235:1935
main error: connection failed: Connection refused
access_realrtsp error: cannot connect to 192.168.31.235:1935
live555 error: Failed to connect with rtsp://192.168.31.235:1935/
main error: connection failed: Connection refused
satip error: Failed to connect to RTSP server 192.168.31.235:1935
main error: connection failed: Connection refused
access_realrtsp error: cannot connect to 192.168.31.235:1935
videotoolbox: Using Video Toolbox to decode 'h264'
main error: buffer deadlock prevented
main error: buffer deadlock prevented
videotoolbox: Using Video Toolbox to decode 'h264'
videotoolbox: vt cvpx chroma: 420v
auhal error: AudioObjectAddPropertyListener failed, device id 135, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 103, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 119, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 91, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 54, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 64, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 60, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 135, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 103, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 119, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 91, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 54, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 64, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 60, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 135, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 103, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 119, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 91, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 54, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 64, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 60, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 135, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 103, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 119, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 91, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 54, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 64, prop: [atfp], OSStatus: 1852797029
auhal error: AudioObjectAddPropertyListener failed, device id 60, prop: [atfp], OSStatus: 1852797029
